### PR TITLE
fix(i18n): update Chinese translation of basic javascript

### PR DESCRIPTION
**Pull request by ZhichengChen**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

### CODE REVIEW

**Comment by RandellDawson** [comment]
@raisedadead Should the non-English languages include the English scrimba url and the English forum topic id as was done on this PR?

**Comment by raisedadead** [comment]
> I think this is not a good idea if we eventually want to have the hints/solutions on the Chinese forum and a Chinese version of each scrimba.

I understand, but as long as we do not have those we should continue with what we have.

File: curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/basic-javascript/add-new-properties-to-a-javascript-object.chinese.md [hunk 1]
```diff
@@ -2,25 +2,35 @@
...
+<code>ourDog.bark = "bow-wow";</code> 
+或者
+<code>ourDog["bark"] = "bow-wow";</code>
+现在当我们访问<code>ourDog.bark</code>时会得到 ourDog 的 bark 值 "bow-wow".
+</section>
 
 ## Instructions
-<section id="instructions">向<code>myDog</code>添加<code>&quot;bark&quot;</code>属性并将其设置为狗声，例如“woof”。您可以使用点或括号表示法。 </section>
+<section id='instructions'>
+给<code>myDog</code>添加一个<code>"bark"</code>属性，设置它的值为狗的声音，例如："woof"。
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/basic-javascript/add-new-properties-to-a-javascript-object.chinese.md]: 句末添加：你可以使用点或中括号操作符。

File: curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/basic-javascript/adding-a-default-option-in-switch-statements.chinese.md [hunk 2]
```diff
@@ -2,37 +2,59 @@
...
+    break;
+}
+```
+
+</section>
 
 ## Instructions
-<section id="instructions">写一个switch语句来设置以下条件的<code>answer</code> ： <br> <code>&quot;a&quot;</code> - “苹果” <br> <code>&quot;b&quot;</code> - “鸟” <br> <code>&quot;c&quot;</code> - “猫” <br> <code>default</code> - “东西” </section>
+<section id='instructions'>
+写一个根据下面的条件来设置<code>answer</code>的switch语句：<br><code>"a"</code> - "apple"<br><code>"b"</code> - "bird"<br><code>"c"</code> - "cat"<br><code>default</code> - "stuff"
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/basic-javascript/adding-a-default-option-in-switch-statements.chinese.md]: 写一个 switch 语句，根据下面的条件来设置<code>answer</code>

File: curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/basic-javascript/appending-variables-to-strings.chinese.md [hunk 3]
```diff
@@ -2,25 +2,30 @@
...
-videoUrl: ''
+videoUrl: 'https://scrimba.com/c/cbQmZfa'
+forumTopicId: 16656
 localeTitle: 将变量附加到字符串
 ---
 
 ## Description
-<section id="description">正如我们可以在字符串<dfn>文字中</dfn>构建多行的字符串一样，我们也可以使用加号等于（ <code>+=</code> ）运算符将变量附加到字符串。 </section>
+<section id='description'>
+我们不仅可以创建出多行的字符串，还可以使用加等号(<code>+=</code>)运算符来追加变量到字符串上。
```
Comment (miyaliu666) [inline curriculum/challenges/chinese/02-javascript-algorithms-and-data-structures/basic-javascript/appending-variables-to-strings.chinese.md]: 还可以使用加等号(<code>+=</code>)运算符来追加变量到字符串上 => 还可以使用加等号(<code>+=</code>)运算符来将变量追加到字符串

**Review by miyaliu666 — COMMENTED** [review]
校对完毕，没有非常详细地校对细节，建议先上线，然后大家之后可以再开 issue 或者 提 pr

**Comment by RandellDawson** [comment]
@ZhichengChen Please address the numerous merge conflicts present on this PR.

Thank you.

**Comment by ZhichengChen** [comment]
@RandellDawson Could you please give me the permission to resolve this PR's conflicts.

**Comment by raisedadead** [comment]
I think you would need to rebase this PR on top of master, which would also mean resolving all the conflicts.

```console
git fetch --all --prune
git checkout i18n-cn-basic-javascript
git pull --rebase upstream master
```

**Carefully** Proceed to fixing all the conflicts and then continue with:

```console
git add .
git rebase --continue
```

Once there are no conflicts, you can:

```console
git push --force origin i18n-cn-basic-javascript
```

**Comment by ZhichengChen** [comment]
@raisedadead Thank you for help, Mrugesh.

**Comment by ZhichengChen** [comment]
@RandellDawson I have addressed the merge conflicts.

**Comment by raisedadead** [comment]
Thanks I am going to land this as is, and we will do a round of audit once again just before we bring this upto speed with master

**Pull request merged**